.github/workflows/build-with-clang.yml

```diff
@@ -52,7 +53,7 @@ jobs:
       - name: Install mkl_random dependencies
         shell: bash -l {0}
         run: |
-          pip install cython setuptools pytest pytest-cov
+          pip install cython setuptools">=70.1" wheel
```

Comment:
Do we need to update `docs/source/maintenance/index.rst` also with min setuptools version required?

Reply:
We don't in dpctl so I don't think it's necessary here